drms/json.py

```diff
@@ -384,18 +397,26 @@ def exp_request(self, ds, notify, method='url_quick', protocol='as-is',
         if filenamefmt is not None:
             d['filenamefmt'] = filenamefmt
 
-        if n is not None:
-            n = int(n)
-            d['process=n'] = '%d' % n
+        n = int(n) if n is not None else 0
+        d['process=n'] = '%d' % n
+        if process is not None:
+            allowed_processes = ['im_patch', 'resize', 'rebin', 'aia_scale_aialev1',
```

Comment:
I see some 'missing' elements in the `allowed_processes` list in _exp_request_url(...). In addition to 'aia_scale_aialev1', there are 'aia_scale_orig' and 'aia_scale_other'. And, as @mbobra has pointed out, 'HmiB2ptr' was also omitted. There is no reason all processing types need be present, but I thought I'd mention ones that might be not well-known.

There is an algorithm used to choose between the three aia_scale variants. First, all variants operate only on certain series. SunPy should disable all variants when any of these conditions are false:

* the series' images are 2D
* keyword CTYPE1 = 'HPLN-TAN'
* keyword CTYPE2 = 'HPLN-TAN'

Second, if aia_scale processing is being performed on aia.lev1, the proper variant is 'aia_scale_orig'. If the processing is being performed on any other aia lev1 series, then the proper variant is 'aia_scale_aialev1'. For all other series where the three restrictions listed above hold, the variant is 'aia_scale_other'.

If the correct variant to use is 'aia_scale_orig', then SunPy should force the value of the filename format variable (`filenamefmt`) to 'aia.lev1.{T_OBS:A}.{WAVELNTH}.{segment}'. Otherwise, the export attempt will fail. To determine if a series is a level-1 series, there needs to be a check for the value of another keyword - KEYWORD_LVL_NUM.

Since choosing the correct variant requires performing this cumbersome algorithm, it might be desirable to simply not support any of the aia_scale processing types. OR, it might be desirable to support the aia_scale_orig variant only, and only if the series is aia.lev1. aia_scale_orig operating on aia.lev1 results in aia.lev1p5 data (level 1.5 aia data).

Likewise, 'HmiB2ptr' operates only under certain conditions, on certain series. I can provide that algorithm if it is wanted, but again, if it is not needed, then I would not bother.

Reply:
Thanks Art! This is really helpful. For some reason, I was thinking I had stuck `HmiB2ptr` in there, but clearly I left it out on accident.

Checking the series name at this point and adjusting the the `aia_scale_*` command should be relatively straightforward. However, I'm not sure about the keyword checks or how to know which series of images are 2D. 

Given all of these complications and of course the possibility of upstream changes to how the JSOC performs these processing operations, I'm hesitant to put really restrictive checks on all processing options. Maybe we should just view this as a flexible, albeit unchecked interface to the processing keywords.

Ideally, users who want a higher level interface to these options will use sunpy (e.g. sunpy/sunpy#4595) and at that stage, we will do much more careful input validation.

Reply:
I've added the additional `aia_scale_*` keywords to the allowed processing key list

Reply:
One half-formed idea I had to consider in the future (I would go ahead with what you have now) would be for the JSOC to provide the list of applicable processing types. Maybe that would get returned to the drms package by the jsoc_info AJAX call. That way the JSOC can evaluate the very-JSOC-specific conditions for the focal series and determine which processing steps to offer. But that does present the issue that the drms client would need to call one of the client API functions that in turn calls the jsoc_info AJAX. The client could not directly call export() without first doing this.

I should note that the JSOC and our export-page and export-script users always invoke jsoc_info first. It is a requirement on the JSOC export web page.

This type of solution, where the drms package can remove itself from some specialized JSOC thing, deferring to the JSOC, is going to be my general suggestion for issues like this. This was the rationale behind my suggestion that the drms package not append the [? quality >= 0 ?] filter to record-set specifications - let the JSOC determine when to add this filter.

Reply:
Ah ok. Yes, that sounds like the most optimal solution and of course would reduce the burden of downstream packages like `drms` if we could use JSOC itself to validate the API. How long do you think the timeline would be for a change like that? Should we go ahead with the solution implemented here and then make the changes to use that information from `jsoc_info` at a later date?

Reply:
Go ahead with what you have now; I don't have an ETA for this - I probably will not be able to get to it for a while.

Reply:
👍 works for me.